File: heeyeon/9주차/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int start = 0;
        int end = height.length -1;
        int w = 0, h = 0;
        int container = 0;
        
        while(start<end){
            w = end - start;
            h = Math.min(height[start], height[end]);
            if(w*h > container)
                container = w*h;
            if(height[end] > height[start]){
                start++;
            }else
                end--;
        }
        return container;
    }
}

File: taewan/week 9/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start, end, sh, eh, ans = 0, len(height) - 1, 0, 0, 0
        
        while start < end:
            if sh > height[start]:
                start += 1
                continue
            
            if eh > height[end]:
                end -= 1
                continue
            
            if height[start] == height[end]:
                sh, eh = height[start], height[end]
                ans = max(ans, sh * (end - start))
                start += 1
                end -= 1
            elif height[start] > height[end]:
                sh, eh = height[start], height[end]
                ans = max(ans, eh * (end - start))
                end -= 1
            else:
                sh, eh = height[start], height[end]
                ans = max(ans, sh * (end - start))
                start += 1
                
        return ans